### ISSUE TITLE
PAD token in seq2seq_attn example

### ISSUE THREAD

**Issue by rickykaka** [asyml/texar#62]
I found PAD token has been counted in the vocab, and take part in the training and decoding process,  does that mean we may generate a PAD token in the middle of our generated sentence?(before generating the EOS token)
This doesn't make sense for me, as in tensorflow nmt tutorial, the pad ID has not been counted in the vocab and not in the final softmax procedure.

**Comment by ZhitingHu** [comment]
As all PAD tokens are added after EOS, a model trained on such data is unlikely to generate PAD before EOS. Thus adding PAD to the vocab has negligible effect.
